### PR TITLE
fix: EF-4518 fixed the fuel consumption formula

### PR DESCRIPTION
**Pull request by 0xxlegolas**
- changed fuelConsumption per minute to behave like oneUnitFuelFuelConsumptionIntervalInSec
- change the tick rate to consume 1 unit of fuel in 18 decimal points
- added a public view function to get the current fuel in wei, so that the float can be calculated offchain to get the accurate results
` currentFuelAmountInWei` 

Ref: https://ccpgames.atlassian.net/browse/EF-4518

### CODE REVIEW

File: mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol [hunk 1]
```diff
@@ -365,7 +378,9 @@ contract SmartDeployable is EveSystem, SmartDeployableErrors {
       entityId
     );
     // elapsed time in seconds, multiplied by consumption rate per second
```
Comment (vayan) [inline mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: nit: outdated comment now

File: mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol [hunk 2]
```diff
@@ -132,7 +137,11 @@ contract SmartDeployable is EveSystem, SmartDeployableErrors {
     _updateFuel(entityId);
     uint256 currentFuel = DeployableFuelBalance.getFuelAmount(_namespace().deployableFuelBalanceTableId(), entityId);
     if (currentFuel < 1) revert SmartDeployable_NoFuel(entityId);
-    DeployableFuelBalance.setFuelAmount(_namespace().deployableFuelBalanceTableId(), entityId, currentFuel - 1); //forces it to tick
+    DeployableFuelBalance.setFuelAmount(
+      _namespace().deployableFuelBalanceTableId(),
+      entityId,
+      currentFuel - (1 * (10 ** DECIMALS))
```
Comment (vayan) [inline mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: We could store `10 ** DECIMALS` in a `ONE_FUEL_UNIT_IN_WEI` const to make the code more readable

Comment (vayan) [reply mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: then for this one it'll be
```suggestion
      currentFuel - ONE_FUEL_UNIT_IN_WEI
```

(can also update all the others)

File: mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol [hunk 3]
```diff
@@ -365,7 +378,9 @@ contract SmartDeployable is EveSystem, SmartDeployableErrors {
       entityId
     );
     // elapsed time in seconds, multiplied by consumption rate per second
-    uint256 fuelConsumed = (block.timestamp - data.lastUpdatedAt) * (data.fuelConsumptionPerMinute / 60);
+    uint256 fuelConsumed = (((block.timestamp - data.lastUpdatedAt) * (10 ** DECIMALS)) /
+      (data.fuelConsumptionPerMinute));
```
Comment (vayan) [inline mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: Is it possible to store `fuelConsumptionPerMinute` in a new `oneFuelUnitConsumptionIntervalInSec` variable at the beginning of ` _currentFuelAmount`? 
And we can add a comment on the assignment to explain why

File: mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol (hunk 2 above)
Comment (0xxlegolas) [reply mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: fixed in [e72f25d](https://github.com/projectawakening/world-chain-contracts/pull/114/commits/e72f25dffac755c0c835ad7ac3f380d680994438)

File: mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol (hunk 3 above)
Comment (0xxlegolas) [reply mud-contracts/world/src/modules/smart-deployable/systems/SmartDeployable.sol]: fixed in [e72f25d](https://github.com/projectawakening/world-chain-contracts/pull/114/commits/e72f25dffac755c0c835ad7ac3f380d680994438)

**Review by vayan — APPROVED** [review]
Seems good to me!

**Review by fc-tezza — APPROVED** [review]

**Pull request merged**